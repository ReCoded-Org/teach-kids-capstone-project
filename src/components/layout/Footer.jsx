// import react from "react";
import logo from '../../images/Logo.png'
import twitter from '../../images/twitter.png'
import linkedin from '../../images/linkedin.png'
import instagram from '../../images/instagram.png'
import facebook from '../../images/facebook.png'
import youtube from '../../images/youtube.png'

export default function Footer(){
  return (
    <div> 
      <div data-testid='footer' className='py-10 flex flex-wrap items-center justify-around bg-primary text-white'>
        <div>
          <img className='pb-2' src={logo} alt='Logo' />
          <h6 className='flex justify-start pt-1'>Events</h6>
          <h6 className='flex justify-start pt-1'>About Us</h6>
          <h6 className='flex justify-start pt-1'>Contact Us</h6>
          <div className='flex pt-8'>
            <img className='pr-8' src={twitter} alt='twitter' />
            <img className='pr-8' src={linkedin} alt='linkedin' />
            <img className='pr-8' src={instagram} alt='instagram' />
            <img className='pr-8' src={facebook} alt='facebook' />
            <img className='pr-8' src={youtube} alt='youtube' />
          </div>

        </div>
        
        <div className='t-10 flex flex-col'>
            <h3 className="flex justify-start text-lg mb-5 pt-10">Subscribe to our newsletter</h3>
            <form >
              <h4 className="flex justify-start ">E-mail</h4>
              <input className='rounded-md text-black' type='text' />
              <button className='bg-redButton p-2 rounded-md'>Submit &#8594;</button>
            </form>
        </div>

      </div>

      <div className='py-1 sm:flex flex-wrap justify-around bg-secondary text-white'>
        <div className='grid grid-cols-1'>
          <p className='my-1 pl-5 justify-self-start text-xs'>All right reserved</p>
        </div>

        <div className='grid grid-cols-1 sm:justify-items-stretch sm:flex sm:justify-between'>
          <div className='my-1 pl-5 text-xs justify-self-start'>Terms and Conditions</div>
          <div className='my-1 pl-5 text-xs justify-self-start'>Privacy and Cookies</div>
          <div className='my-1 pl-5 pr-2 text-xs justify-self-start'>Accessability</div>
        </div>
      </div>
    </div>
  )
}