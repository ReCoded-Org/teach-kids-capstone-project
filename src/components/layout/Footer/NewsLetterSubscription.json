{
  "Users": [
    {
      "email": "mustafaRast@g.com",
      "id": 1
    },
    {
      "email": "hi@hi.hi",
      "id": 2
    }
  ]
}