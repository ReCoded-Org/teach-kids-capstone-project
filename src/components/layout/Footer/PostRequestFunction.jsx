import React from "react";

function PostRequestFunction(value) {
    const data = {
        email: value,
    };

    fetch("http://localhost:3000/Users", {
        method: "POST",
        headers: {
            "Content-Type": "application/json",
        },
        body: JSON.stringify(data),
    })
        .then((res) => res.json())
        .then((data) => console.log(data));
}

export default PostRequestFunction;
