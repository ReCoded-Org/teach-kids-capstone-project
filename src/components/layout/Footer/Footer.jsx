import react, { useState } from "react";
import logo from "../../../assets/Logo.png";
import twitter from "../../../assets/twitter.svg";
import linkedin from "../../../assets/linkedin.svg";
import instagram from "../../../assets/instagram.svg";
import facebook from "../../../assets/facebookFooter.svg";
import youtube from "../../../assets/youtube.svg";
import PostRequestFunction from "./PostRequestFunction";
import { isValidEmail } from "../../../services/ValidationFunctions";

export default function Footer() {
    const [subscriptionData, setSubscriptionData] = useState("");
    const [validationMsg, setValidationMsg] = useState("");
    return (
        <div>
            <div
                data-testid='footer'
                className='jus flex flex-wrap items-center bg-primary px-12 py-10 font-SourceSansPro text-xl text-white md:justify-around lg:justify-between lg:px-36'
            >
                <div className='my-2'>
                    <img className='pb-2' src={logo} alt='Logo' />
                    <h6 className='flex justify-start pt-2'>Events</h6>
                    <h6 className='flex justify-start pt-2'>About Us</h6>
                    <h6 className='flex justify-start pt-2'>Contact Us</h6>
                    <div className='flex pt-8'>
                        <img className='pr-8' src={twitter} alt='twitter' />
                        <img className='pr-8' src={linkedin} alt='linkedin' />
                        <img className='pr-8' src={instagram} alt='instagram' />
                        <img className='pr-8' src={facebook} alt='facebook' />
                        <img className='pr-8' src={youtube} alt='youtube' />
                    </div>
                </div>
                <div className='t-10 flex flex-col lg:mr-6'>
                    <h3 className='mb-5 flex justify-start pt-10 text-lg'>
                        Subscribe to our newsletter
                    </h3>
                    <div className='flex justify-between'>
                        <h4 className=' '>E-mail</h4>
                        <h4 className=' mx-6 text-red'>{validationMsg}</h4>
                    </div>
                    <form>
                        <h4 className='flex justify-start'>E-mail</h4>
                        <input
                            className='rounded-md text-black'
                            type='text'
                            onChange={(e) => {
                                setSubscriptionData(e.target.value);
                                setValidationMsg("");
                            }}
                            value={subscriptionData}
                        />
                        <button
                            className='transform rounded-md border-2 border-red bg-red p-1.5 duration-300 ease-linear hover:border-white hover:bg-blue-light hover:shadow'
                            onClick={(e) => {
                                e.preventDefault();
                                if (isValidEmail(subscriptionData)) {
                                    PostRequestFunction(subscriptionData);
                                } else {
                                    setValidationMsg("Enter a Valid Email");
                                }
                            }}
                        >
                            Submit &#8594;
                        </button>
                    </form>
                </div>
            </div>
            <div className='flex-wrap justify-around bg-secondary pb-2 pt-1 text-white sm:flex'>
                <div className='grid grid-cols-1'>
                    <p className='my-1 justify-self-start pl-5 text-xs'>
                        All right reserved
                    </p>
                </div>
                <div className='grid grid-cols-1 sm:flex sm:justify-between sm:justify-items-stretch'>
                    <div className='my-1 justify-self-start pl-5 text-xs'>
                        Terms and Conditions
                    </div>
                    <div className='my-1 justify-self-start pl-5 text-xs'>
                        Privacy and Cookies
                    </div>
                    <div className='my-1 justify-self-start pl-5 pr-2 text-xs'>
                        Accessability
                    </div>
                </div>
            </div>
        </div>
    );
}
