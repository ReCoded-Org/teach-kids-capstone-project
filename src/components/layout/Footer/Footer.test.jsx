import {render} from '@testing-library/react';
import Footer from './Footer';

describe("Footer Component Test", () => {

  it("rendered footer", () => {

    const {getByTestId} = render(<Footer />);
    const footer = getByTestId("footer")
    expect(footer).toBeTruthy()

  });

});

