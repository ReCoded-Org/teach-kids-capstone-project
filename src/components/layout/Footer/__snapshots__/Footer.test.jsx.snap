// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="bg-primary"
>
  <div
    className="mx-auto flex w-9/12 flex-wrap items-center justify-between bg-primary py-10  font-SourceSansPro text-xl text-white md:px-8  lg:justify-between "
    data-testid="footer"
  >
    <div
      className="my-2"
    >
      <img
        alt="Logo"
        className="pb-2"
        src="Logo.png"
      />
      <a
        href="/events"
        onClick={[Function]}
      >
        <h6
          className="flex justify-start pt-2"
        >
          footer.pages.0
        </h6>
      </a>
      <a
        href="/#contact-us"
        onClick={[Function]}
      >
        <h6
          className="flex justify-start pt-2"
        >
          footer.pages.1
        </h6>
      </a>
      <a
        href="/#about"
        onClick={[Function]}
      >
        <h6
          className="flex justify-start pt-2"
        >
          footer.pages.2
        </h6>
      </a>
      <a
        href="/our-team"
        onClick={[Function]}
      >
        <h6
          className="flex justify-start pt-2"
        >
          footer.pages.3
        </h6>
      </a>
      <div
        className="flex pt-8"
      >
        <img
          alt="twitter"
          className="pr-8"
          src="twitter.svg"
        />
        <img
          alt="linkedin"
          className="pr-8"
          src="linkedin.svg"
        />
        <img
          alt="instagram"
          className="pr-8"
          src="instagram.svg"
        />
        <img
          alt="facebook"
          className="pr-8"
          src="facebookFooter.svg"
        />
        <img
          alt="youtube"
          className="pr-8"
          src="youtube.svg"
        />
      </div>
    </div>
    <div
      className="t-10 flex flex-col"
    >
      <h3
        className="mb-5 flex justify-start pt-10 text-2xl"
      >
        footer.newsletter.title
      </h3>
      <div
        className="flex justify-between"
      >
        <h4
          className=" mx-6 text-red"
        />
      </div>
      <form>
        <h4
          className="flex justify-start"
        >
          footer.newsletter.email
        </h4>
        <input
          className="rounded-md text-black"
          onChange={[Function]}
          type="text"
          value=""
        />
        <button
          className="transform rounded-md border-2 border-red bg-red p-1.5 duration-300 ease-linear hover:border-white hover:bg-blue-light hover:shadow"
          onClick={[Function]}
        >
          footer.newsletter.btn
           â†’
        </button>
      </form>
    </div>
  </div>
  <div
    className="bg-secondary"
  >
    <div
      className="mx-auto flex w-9/12 flex-col flex-wrap justify-start  pb-2 pt-1 text-white md:flex md:flex-row md:justify-between"
    >
      <div
        className="grid grid-cols-1"
      >
        <p
          className="my-1 justify-self-start md:pl-5 text-xs"
        >
          footer.lowerFooter.0
        </p>
      </div>
      <div
        className="grid grid-cols-1 sm:flex sm:justify-between sm:justify-items-stretch"
      >
        <div
          className="my-1 justify-self-start md:pl-5 text-xs"
        >
          footer.lowerFooter.1
        </div>
        <div
          className="my-1 justify-self-start md:pl-5 text-xs"
        >
          footer.lowerFooter.2
        </div>
        <div
          className="my-1 justify-self-start md:pl-5 pr-2 text-xs"
        >
          footer.lowerFooter.3
        </div>
      </div>
    </div>
  </div>
</div>
`;
